### ISSUE TITLE
incorrect values from `mongodbatlas_cluster` and `mongodbatlas_clusters` datasources

### ISSUE THREAD

**Issue by swapzero** [mongodb/terraform-provider-mongodbatlas#618]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v0.12.31
+ provider.mongodbatlas v1.1.1
```

### Terraform Configuration File 

```hcl
data "mongodbatlas_project" "playground" {
  name = "project_name"
}

data "mongodbatlas_cluster" "playground" {
  project_id = data.mongodbatlas_project.playground.id
  name = "cluster_name"
}

/* ...or get ALL the clusters
data "mongodbatlas_clusters" "playground" {
  project_id = data.mongodbatlas_project.playground.id
}
*/
```

### Steps to Reproduce

- target project has **only one cluster**
- current auto-scaling settings for the cluster
![image](https://user-images.githubusercontent.com/45405723/142844788-8a5805f4-f8a7-499a-94d7-32ea83b317e5.png)

### Expected Behavior

I have called the API directly and, **based on the current data returned by Atlas API**  , the datasources should return the following
```
"auto_scaling_compute_enabled": true,
"auto_scaling_compute_scale_down_enabled": false,
"auto_scaling_disk_gb_enabled": true,
"provider_auto_scaling_compute_max_instance_size": "M80",
"provider_auto_scaling_compute_min_instance_size": null,
```
Relevant data returned by the API call
```
  "autoScaling": {
...
    "compute": {
      "enabled": true,
      "scaleDownEnabled": false
    },
    "diskGBEnabled": true
  },
```
and
```
  "providerSettings": {
...
    "autoScaling": {
      "compute": {
        "maxInstanceSize": "M80",
        "minInstanceSize": null
      }
    },
```
ℹ️ Not sure why minInstanceSize is `null` here but it has nothing to do with this issue.

### Actual Behavior

- `mongodbatlas_cluster` datasource returns
```
"auto_scaling_compute_enabled": null,
"auto_scaling_compute_scale_down_enabled": null,
"auto_scaling_disk_gb_enabled": true,
"provider_auto_scaling_compute_max_instance_size": null,
"provider_auto_scaling_compute_min_instance_size": null,
```
- `mongodbatlas_clusters` datasource returns
```
"auto_scaling_compute_enabled": false,
"auto_scaling_compute_scale_down_enabled": false,
"auto_scaling_disk_gb_enabled": false,
"provider_auto_scaling_compute_max_instance_size": "",
"provider_auto_scaling_compute_min_instance_size": "",
```

**Comment by nikhil-mongo** [comment]
@swapzero I shall do the repro and get back to you.

**Comment by nikhil-mongo** [comment]
@swapzero I could repro what you mentioned here in this issue.
I am checking internally if this is expected behaviour that we skip to fetch the optional attributes in data sources and shall update you when I have more information.

Thanks

**Comment by themantissa** [comment]
Internal ticket: INTMDB-280

**Comment by themantissa** [comment]
@swapzero we are about to release the fix in version 1.2.  The pre-release is completed if you'd like to test first.  Thank you.

**Comment by themantissa** [comment]
Fixed in 1.2
